src/VisualStudio/Core/Def/ColorSchemes/ColorSchemeApplier.cs

```diff
@@ -142,33 +129,36 @@ private bool TryGetUpdatedColorScheme([NotNullWhen(returnValue: true)] out Color
                 : ColorSchemeName.VisualStudio2017;
 
             if (appliedColorScheme == configuredColorScheme)
-            {
-                colorScheme = null;
-                return false;
-            }
+                return null;
 
-            colorScheme = configuredColorScheme;
-            return true;
+            return configuredColorScheme;
         }
 
         public bool IsSupportedTheme()
-            => IsSupportedTheme(GetThemeId());
+            => _threadingContext.JoinableTaskFactory.Run(() => IsSupportedThemeAsync(_threadingContext.DisposalToken));
 
-        public bool IsSupportedTheme(Guid themeId)
+        private async Task<bool> IsSupportedThemeAsync(CancellationToken cancellationToken)
+            => IsSupportedTheme(await GetThemeIdAsync(cancellationToken).ConfigureAwait(false));
+
+        private bool IsSupportedTheme(Guid themeId)
         {
             return _colorSchemes.Values.Any(
                 scheme => scheme.Themes.Any(
                     theme => theme.Guid == themeId));
         }
 
         public bool IsThemeCustomized()
-            => _classificationVerifier.AreForegroundColorsCustomized(_settings.GetConfiguredColorScheme(), GetThemeId());
+            => _threadingContext.JoinableTaskFactory.Run(async () =>
+                await _classificationVerifier.AreForegroundColorsCustomizedAsync(
+                    _settings.GetConfiguredColorScheme(),
+                    await GetThemeIdAsync(_threadingContext.DisposalToken).ConfigureAwait(false),
+                    _threadingContext.DisposalToken).ConfigureAwait(false));
```

Comment:
You could get rid of the .ConfigureAwait's here since we're in a definitely JTF safe situation. Our analyzers might not like it though. (Wonder if we should teach them to ignore that in JTF.Runs...)

Reply:
done.